python/cudf/cudf/io/parquet.py

```diff
@@ -577,11 +577,44 @@ def read_parquet(
         )
     filepath_or_buffer = paths if paths else filepath_or_buffer
 
+    # Prepare remote-IO options
+    prefetch_options = kwargs.pop("prefetch_options", {})
+    if not ioutils._is_local_filesystem(fs):
+        method = prefetch_options.get("method")
+        _row_groups = None
+        if method in (None, "parquet"):
```

Comment:
The idea here is to use `"parquet"` by default if we are only reading a subset of row groups, and to otherwise use `"all"` as the default. I am using `None` to signify that the user has not specified anything, so we can choose what is likely to perform best.

In my experience, `"all"` is more performant unless you are only reading a small part of each file. For example, when you break up large parquet files with dask, and each partition is being created from a small fraction of a single file. Using `"all"` in that case will duplicate a lot of work. Using `"parquet"` would also be more performant if you are reading a small subset of columns, **unless** that small subset includes a large text column (which is pretty common in NeMo Curator).

For this particular iteration, I decided it was best to use `"all"` in most cases, but to allow the user to opt in to "parquet" as long as the row-group selection matches for every file included in the read (a requirement of `fsspec_parquet._get_parquet_byte_ranges`).